lib/msfdb_helpers/pg_ctl.rb

```diff
@@ -28,6 +40,37 @@ def init(msf_pass, msftest_pass)
       restart
     end
 
+    # Creates and attempts to execute a testfile in the specified directory,
+    # to determine if it is mounted with NOEXEC flags.
+    def test_executable_file(path)
+      begin
+        file_name = File.join(path, 'msfdb_testfile')
+        File.open(file_name, 'w') do |f|
+          f.puts "#!/bin/bash\necho exec"
+        end
+        File.chmod(0744, file_name)
+        
+        if run_cmd(file_name)
+          File.open("#{@db}/postgresql.conf", 'a') do |f|
+            f.puts "unix_socket_directories = \'#{path}\'"
+          end
```

Comment:
Will this append take preference over any previous entries or will the first instance take preference? Would we maybe want to read the whole file in here and do a regex instead to replace things?

Reply:
Further testing showed that this was not a problem in my tests, so marking this as resolved.